build-system/compile/log-messages-formats.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const pathPrefix = 'dist/log-messages';
+
+/**
+ * Source of truth for extracted messages during build, but should not be
+ * deployed. Shaped `{message: {id, message, ...}}`.
+ */
+const extractedPath = `${pathPrefix}.by-message.json`;
```

Comment:
Moved extractedPath and formats from log-messages.js into this new file because of a transitive dependency issue from the transform tests caused by using endBuildStep() as suggested here. Whew.